### PR TITLE
Stop users from giving themselves karma

### PR DESCRIPTION
**Pull request by himmelmaus**
Similar to previous PR, test name with "~d" at the end - except should pass the linter this time

### CODE REVIEW

**Comment by neanias** [comment]
For future reference, you can add and push another commit and the CI will auto-build it again.

File: ircbot/plugins/karma.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def message_hook(bot, channel, sender, message):
         term = m.group('term')
         reason = m.group('reason')
         term = term.strip().lower()
-        if term == sender.lower():
+        if term == sender.lower() or term + "~d" == sender.lower():
```
Comment (neanias) [inline ircbot/plugins/karma.py]: The CI hasn't run yet, but I'm not sure how the + associates.

**Comment by neanias** [comment]
@findoslice, have you tested this?

**Comment by himmelmaus** [comment]
@neanias tested as to whether it works as a patch or with the linter? If the former, I have a bot with the similar karma configuration and on a shared discord/irc channel where it works so to my knowledge it will.

**Comment by neanias** [comment]
I was wondering whether the (+) has higher precedence than (==)

**Comment by himmelmaus** [comment]
Oh, in that case it definitely does (tested that myself right now to be sure)

**Comment by neanias** [comment]
Fair play then. See follow up discussion on IRC

**Review by neanias — APPROVED** [review]

**Comment by neanias** [comment]
@hreeder Got any opinions?

**Comment by hreeder** [comment]
I'd probably like to see the suffix moved to the config file, so it's something we can expand in future (say we have to recognize users from another service that come in with ~s on the end?)

In that event, maybe it becomes a regex where we insert `term` dynamically? Open to ideas on how to achieve that.

**Comment by hreeder** [comment]
Thinking about it, if we went down that route of having some kind of configurable regex, we could drop the or entirely, because we could perform that or in the regex

**Comment by neanias** [comment]
Regex++

**Comment by hreeder** [comment]
Looking better, @findoslice - now I think just move it out to `bot.config` (maybe `bot.config['Karma']['self_re']`)

**Comment by hreeder** [comment]
LGTM :+1: 

Thanks @findoslice

**Pull request merged**